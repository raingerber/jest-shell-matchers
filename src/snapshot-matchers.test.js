const extendJestWithShellMatchers = require('../index')

beforeAll(() => {
  extendJestWithShellMatchers()
})

describe('snapshot matchers', () => {
  const input = [
    'sh',
    ['./test-scripts/input-output.sh'], {
      env: {
        TO_STDOUT: `The glorious gods
  sit in hourly synod about thy
      particular prosperity
`,
        TO_STDERR: `A noble fellow,
        I warrant him.`,
      }
    }
  ]
  it('toHaveSpawnOutputMatchingSnapshot', async () => {
    await expect(input).toHaveSpawnOutputMatchingSnapshot()
  })
  it('toHaveStdoutMatchingSnapshot', async () => {
    await expect(input).toHaveStdoutMatchingSnapshot()
  })
  it('toHaveStderrMatchingSnapshot', async () => {
    await expect(input).toHaveStderrMatchingSnapshot()
  })
})

