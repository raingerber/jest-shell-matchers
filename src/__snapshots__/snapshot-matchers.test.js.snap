// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot matchers toHaveSpawnOutputMatchingSnapshot: toHaveSpawnOutputMatchingSnapshot 1`] = `
Object {
  "code": 0,
  "signal": "",
  "stderr": "A noble fellow, I warrant him.
",
  "stdout": "The glorious gods sit in hourly synod about thy particular prosperity
",
}
`;

exports[`snapshot matchers toHaveStderrMatchingSnapshot: toHaveStderrMatchingSnapshot 1`] = `
"A noble fellow, I warrant him.
"
`;

exports[`snapshot matchers toHaveStdoutMatchingSnapshot: toHaveStdoutMatchingSnapshot 1`] = `
"The glorious gods sit in hourly synod about thy particular prosperity
"
`;
